lib/services/provided/autocomplete.js

```diff
@@ -145,19 +146,32 @@ export function provideAutocompleteResults(
       const kernel = store.kernel;
       if (!kernel || kernel.executionState !== "idle") return null;
 
-      return new Promise(resolve => {
-        kernel.inspect(replacedText, replacedText.length, ({ data, found }) => {
-          if (!found || !data["text/plain"]) resolve(null);
-          resolve({
+      const promise = new Promise(resolve => {
+        kernel.inspect(replacedText, replacedText.length, ({ found, data }) => {
+          if (!found || !data["text/plain"]) {
+            return resolve(null);
```

Comment:
Without `return` statement here, this callback function will continue to execute if `!found || !data["text/plain"]`, and it would cause Atom crash.